### PR TITLE
WooCommerce: Add the products list UI.

### PR DESCRIPTION
**Pull request by justinshreve**
This PR implements the UI for the products list for v1, which is a list table with pagination.
This PR does not implement searching per the mockup, which still needs to be worked on.

To Test:
* Go to `http://calypso.localhost:3000/store/products/:site`
* See your products load.
* Page through your products.

cc @kellychoffman @jameskoster Since this PR contains UI.

screen shot 2017-06-16 at 11 32 12 am

### CODE REVIEW

**Comment by jameskoster** [comment]
Made a couple of minor tweaks. 

Are we using a table to save time? I'd like to make the entire row a link but we can't with a table. If there's an appropriate list component we could use that would be preferable.

**Comment by justinshreve** [comment]
I was using these components because they were built for this purpose just the other week: https://github.com/Automattic/wp-calypso/pull/14892. Considering we are a week out, and this is tabular data, I see no problem. I still still be able to make the row clickable though.

**Comment by justinshreve** [comment]
@jameskoster The row is now clickable. I added a hover cover for the row, but feel free to adjust.

**Comment by kellychoffman** [comment]
>  The row is now clickable. I added a hover cover for the row, but feel free to adjust.

Made some tweaks to the hover state. 

👍  Looking snazzy!!!

File: client/extensions/woocommerce/app/products/index.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+	renderList() {
+		const { site, products, totalProducts, currentPageLoaded, requestedPageLoaded, requestedPage } = this.props;
+
+		if ( currentPageLoaded === true && totalProducts === 0 ) {
+			return this.renderEmptyContent();
+		}
+
+		let isRequesting = false;
+		if ( requestedPage && ! requestedPageLoaded ) {
```
Comment (belcherj) [inline client/extensions/woocommerce/app/products/index.js]: is the else if needed?  why not just do it in one line with an ||?

File: client/extensions/woocommerce/app/products/products-list-row.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	const renderImage = () => (
+		<div className={ imageClasses }>
+			{ featuredImage && ( <img src={ featuredImage.src } /> ) }
+		</div>
+	);
+
+	const categoryNames = product.categories && product.categories.map( function( category ) {
+		return category.name;
+	} );
+	const renderCategories = () => (
```
Comment (belcherj) [inline client/extensions/woocommerce/app/products/products-list-row.js]: add extra whitespace line before here

File: client/extensions/woocommerce/app/products/products-list-row.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		<div className="products__list-categories">
+			{ categoryNames &&
+				( categoryNames.join( ', ' ) ) ||
+				( <span>{ translate( '-' ) }</span> )
```
Comment (belcherj) [inline client/extensions/woocommerce/app/products/products-list-row.js]: do you translate dashes?

File: client/extensions/woocommerce/app/products/products-list-table.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ */
+import ProductsListRow from './products-list-row';
+import Table from 'woocommerce/components/table';
+import TableRow from 'woocommerce/components/table/table-row';
+import TableItem from 'woocommerce/components/table/table-item';
+
+const ProductsListTable = ( { translate, products, site, isRequesting } ) => {
+	const headings = (
+		<TableRow isHeader className={ classNames( { 'products__list-placeholder': ! products } ) }>
+			{ [ translate( 'Product' ), translate( 'Inventory' ), translate( 'Category' ) ].map( ( item, i ) =>
```
Comment (belcherj) [inline client/extensions/woocommerce/app/products/products-list-table.js]: i like how you did this!

File: client/extensions/woocommerce/app/products/products-list-row.js (hunk 3 above)
Comment (justinshreve) [inline client/extensions/woocommerce/app/products/products-list-row.js]: Nope. We had `N/A` before and didn't remove the wrapper when it changed 😀.

**Pull request merged**